src/gsy_e/models/strategy/future/strategy.py

```diff
@@ -92,11 +96,16 @@ def event_market_cycle(self, strategy: "BidEnabledStrategy") -> None:
     def event_tick(self, strategy: "BidEnabledStrategy") -> None:
         """Base class method for handling the tick"""
 
+    def update_and_populate_price_settings(self, strategy: "BidEnabledStrategy") -> None:
```

Comment:
I can see that all strategies type hints throughout the module are set to `BidEnabledStrategy` when they can also be `OfferEnabledStrategy`, so for these usages I suppose it's safe to use `BaseStrategy`
```suggestion
    def update_and_populate_price_settings(self, strategy: "BaseStrategy") -> None:
```